src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -31,7 +32,33 @@ public List<BookDto> findAll() {
 
     @Override
     public BookDto getBookById(Long id) {
-        Book bookById = bookRepository.getBookById(id);
-        return bookMapper.toDto(bookById);
+        return bookRepository.findById(id).stream()
+                .map(bookMapper::toDto)
+                .findFirst()
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "can`t find book with id: " + id));
+    }
+
+    @Override
+    public BookDto updateBookById(Long id, CreateBookRequestDto updatedBookDto) {
```

Comment:
```suggestion
    public BookDto updateBookById(Long id, UpdateBookRequestDto updatedBookDto) {
```

Recently I discovered a better approach for naming methods and parameters. I believe it would be beneficial to create a new class named UpdateBookRequestDto. Using `CreateBookRequestDto `might be confusing in the future, because exactly we not create anithing here, just update.

Reply:
Or just `BookRequestDto`  since it is used for both create and update 😉